packages/react-scripts/fixtures/kitchensink/jsconfig.json

```diff
@@ -0,0 +1,5 @@
+{
+  "compilerOptions": {
+    "baseUrl": "src"
```

Comment:
I know that projects like Storybook rely on the fact that we didn't have this set previously, and I fear that other users may have too.

I've seen configs set up like this before, which also works.
```
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "*": [ "src/*" ]
    }
  }
```

If we go down the path of setting `baseUrl` to `src`, should we consider having the `tsconfig` file in the root of `src` instead?

Reply:
Since it is a breaking change would it not be better to force these imports to be under `"src/"`, e.g. `import MyAmazingComponent from "src/MyAmazingComponent"` to avoid ambiguities. I guess a way around this would be to place all your code in a subfolder of `src`.

Reply:
@mrmckeb how exactly do they rely on that? Currently neither `baseUrl` nor `paths` are supported in CRA so I'm not sure how we could break anything with this change.

Reply:
I think it can be quite tricky to rebuild the exact same module resolution logic in webpack & jest as in TypeScript if we have to support an array of paths. And the benefits of that are not really clear to me.

Reply:
Can I confirm, that with this approach. a user can define a root `__mocks__` folder and use TypeScript within it? And similarly, a root `.storybook` folder and use TypeScript within it?

This is my concern, if this works - I'm 100% OK with this approach.